core/audits/largest-contentful-paint-element.js

```diff
@@ -134,7 +134,13 @@ class LargestContentfulPaintElement extends Audit {
       settings: context.settings, URL: artifacts.URL};
 
     const elementTable = this.makeElementTable(artifacts);
-    if (!elementTable) return {score: null, notApplicable: true};
+    if (!elementTable) {
+      return {
+        score: null,
+        notApplicable: true,
+        metricSavings: {LCP: 0},
```

Comment:
I've generally been of the mindset that we should avoid adding any of the usual stuff to audits in the `notApplicable` case. It's not a hard and fast rule, but when looking at an LHR, you generally shouldn't have to wonder if a value on an audit is important. The whole thing has been marked not applicable; it's not that there's no savings to be had, it's that this audit is irrelevant for this particular page load.

(the only exception might be `debugdata` if we ever want to pipe through information on why something was marked n/a to http archive or wherever, but I don't know if that's ever come up before)

Reply:
True, but I was planning for `metricSavings` to eventually replace our current metric applicability lists. Removing `metricSavings` when an audit is N/A means that it will always be hidden when the user filters to any metric.

Not necessarily a bad UX, but it's also not how we do the filters right now. Knowing that an audit *can* affect a certain metric even when it's N/A can be useful information IMO.